java-design-patterns/src/main/java/org/example/Main.java

```diff
@@ -17,7 +18,29 @@ private static void testSingleton(){
 
 //        SingletonCache singletonCache2 = globalCache.clone()
     }
+
+    private static CalculatorCreator getCalculatorCreator(String os){
+        if (os == "ubuntu"){
+            return new UbuntuCalculatorCreator();
+        } else if (os == "windows") {
+            return new WindowsCalculatorCreator();
+        }
```

Comment:
one of the objective of the assignment is to avoid changing the Factory class. if we add a new os (Centos) we have to add new condition. 
we can use reflection like `Class.forName("...").getDeclaredConstructor().newInstance()` to get instance